pkg/image/api/helper.go

```diff
@@ -553,8 +553,14 @@ func LatestImageTagEvent(stream *ImageStream, imageID string) (string, *TagEvent
 			continue
 		}
 		for i, event := range events.Items {
-			if event.Image != imageID {
-				continue
+			if d, err := digest.ParseDigest(event.Image); err == nil {
+				if !strings.HasPrefix(d.Hex(), imageID) && !strings.HasPrefix(event.Image, imageID) {
```

Comment:
we use this same digest and `event.Image` logic (and the item below) in the `ResolveImageId` in this same file.  Time for a helper method?